scylla/src/transport/connection.rs

```diff
@@ -102,11 +105,44 @@ pub(crate) enum ConnectionError {
     TranslationError(#[from] TranslationError),
     #[error(transparent)]
     BrokenConnection(#[from] BrokenConnectionError),
+    #[error(transparent)]
+    ConnectionSetupRequestError(#[from] ConnectionSetupRequestError),
     // TODO: remove it or change it later.
     #[error(transparent)]
     QueryError(#[from] QueryError),
 }
 
+#[derive(Error, Debug)]
+#[error("Failed to perform a connection setup request. Request: {request_name}, reason: {kind}")]
+pub(crate) struct ConnectionSetupRequestError {
+    request_name: &'static str,
+    kind: ConnectionSetupRequestErrorKind,
+}
```

Comment:
I don't think I understand this comment. What other type should we use here?

Reply:
Renamed the fields:
`request_name` -> `request_kind`
`kind` -> `error`

Reply:
Unresolved so Karol can explain what doubt he had about &str.

Reply:
My comment was very unclear, sorry. What I meant is why use str instead of enum? One of the main reasons for this whole refactor is to de-stringify errors.

Reply:
On a second thought, I believe the enum is worth the boilerplate.